parser.go

```diff
@@ -1507,103 +1505,145 @@ func (p *parser) getTypeSpec(pkgPath, pkgName, typeName string) (*ast.TypeSpec,
 	return astTypeSpec, true
 }
 
-func (p *parser) parseSchemaPropertiesFromStructFields(pkgPath, pkgName string, structSchema *SchemaObject, astFields []*ast.Field) {
-	if astFields == nil {
-		return
-	}
-	structSchema.Properties = orderedmap.New()
-	if structSchema.DisabledFieldNames == nil {
-		structSchema.DisabledFieldNames = map[string]struct{}{}
+func (p *parser) parseAstFields(pkgPath, pkgName string, structSchema *SchemaObject, astFields []*ast.Field) {
+	for _, astField := range astFields {
+		p.parseAstField(pkgPath, pkgName, structSchema, astField)
 	}
+}
 
-	for _, astField := range astFields {
-		fieldSchema := &SchemaObject{}
-		typeAsString := p.getTypeAsString(astField.Type)
-		typeAsString = strings.TrimLeft(typeAsString, "*")
-		isSliceOrMap := strings.HasPrefix(typeAsString, "[]") || strings.HasPrefix(typeAsString, "map[]")
-		isInterface := strings.HasPrefix(typeAsString, "interface{}")
-		if isSliceOrMap || isInterface || typeAsString == "time.Time" {
+func (p *parser) parseAstField(pkgPath, pkgName string, structSchema *SchemaObject, astField *ast.Field) {
+	fieldSchema := &SchemaObject{}
+	typeAsString := p.getTypeAsString(astField.Type)
+	isSliceOrMap := strings.HasPrefix(typeAsString, "[]") || strings.HasPrefix(typeAsString, "map[]")
+	isInterface := strings.HasPrefix(typeAsString, "interface{}")
+	if isSliceOrMap || isInterface || typeAsString == "time.Time" {
+		splitType := strings.Split(typeAsString, "]")
+		if len(splitType) > 1 {
+			valueType := splitType[1]
+			if !isBasicGoType(valueType) {
```

Comment:
Since the `else` block of both this and the preceding `if` are identical, you should be able to combine these to reduce nesting and DRY the elses, e.g. something like

```
	if isSliceOrMap || isInterface || typeAsString == "time.Time" {
		splitType := strings.Split(typeAsString, "]")
		if len(splitType) > 1 && !isBasicGoType(splitType[1]) {
			if _, ok := p.KnownIDSchema[splitType[1]]; ok {
			... etc
		} else {
			var err error
			fieldSchema, err = p.parseSchemaObject(pkgPath, pkgName, typeAsString, true)
			... etc
		}
	} else if !isBasicGoType(typeAsString) {
```